Python/setup.py

```diff
@@ -61,5 +63,12 @@
       package_data = package_data,
       install_requires = ['numpy', 'picmistandard==0.0.20', 'periodictable'],
       python_requires = '>=3.7',
+      ext_modules=[
+          Extension(
```

Comment:
This extension should only be defined when the installation is including the compiled library, i.e. when `args.with_libwarpx` or `args.with_lib_dir or PYWARPX_LIB_DIR` is true. When those are not true, the setup.py installs a pure Python package. You can do (after line 54)

```
ext_modules = []
if package_data:
    ext_modules.append(Extension(...))
```

and then, `ext_modules = ext_modules` in the call to setup.